collections/soldiers.js

```diff
@@ -33,4 +33,27 @@ function newSoldier(db, data, callback) {
   }
 }
 
-module.exports.newSoldier = newSoldier;
\ No newline at end of file
+function getSoldier(db, name, id, callback) {
+  const collection = db.collection('soldiers');
+  let searchedParams = {};
+
+  if (name) {
```

Comment:
if name is not defined, then assigning searchedParams["name"] = name; would put undefined in it. which is essentially the same as checking and then assigning. the end result of this:

if (name) {
    searchedParams["name"] = name;
  }
  if (id) {
    searchedParams["id"] = id;
  }

is the same as this 

searchedParams["name"] = name;
searchedParams["id"] = id;